lib/cypress/expected_results_calculator.rb

```diff
@@ -73,21 +76,69 @@ def aggregate_results_for_measure(measure, individual_results = nil)
         end
         # extract the observed value from an individual results.  Observed values are in the 'episode result'.
         # Each episode will have its own observation
-        observ_values[key].concat get_observ_values(individual_result['episode_results']) if individual_result['episode_results']
+        next unless individual_result['episode_results']
+
+        collect_observations(observation_hash, individual_result, measure)
       end
-      # TODO: Observations may not always be the median
       @measure_result_hash[measure.hqmf_id].keys.each do |key|
-        @measure_result_hash[measure.hqmf_id][key]['OBSERV'] = observ_values[key] ? median(observ_values[key].reject(&:nil?)) : 0.0
+        calculate_observation(observation_hash, measure, key)
         @measure_result_hash[measure.hqmf_id][key]['measure_id'] = measure.hqmf_id
         @measure_result_hash[measure.hqmf_id][key]['pop_set_hash'] = measure.population_set_hash_for_key(key)
       end
     end
+
+    # adds the observation values found in an individual_result to the observation_hash
+    def collect_observations(observation_hash, individual_result, measure)
+      key = individual_result['population_set_key']
+      population_set = measure.population_set_for_key(key).first
+      # collect the observation_statments for the population_set. There may be more than one. episode_results are recorded in the same order
+      observation_statments = population_set.observations.map { |obs| obs.observation_parameter.statement_name }
+      # collect the observation_values from and individual_result
+      # a scenario with multiple episodes and multiple observations would look like this [[2], [9, 1]]
+      observation_values = get_observ_values(individual_result['episode_results']).compact
+      observation_values.each do |observation_value|
+        observation_value.each_with_index do |observation, index|
+          # lookup the population code (e.g., DENOM is the population code for the statement named 'Denominator')
+          obs_pop = measure.population_keys.find { |pop| population_set.populations[pop]['statement_name'] == observation_statments[index] }
+          # create an empty hash value if one doesn't already exist
+          observation_hash[key][obs_pop] = { values: [], statement_name: observation_statments[index] } unless observation_hash[key][obs_pop]
+          # add the observation to the hash
+          observation_hash[key][obs_pop][:values] << observation
+        end
+      end
+    end
+
+    def calculate_observation(observation_hash, measure, population_set_key)
```

Comment:
there's some good commenting in the function. Maybe just a quick function header comment would still be helpful though?